app/models/postable.rb

```diff
@@ -180,16 +215,85 @@ def pre_post
                         quantity: amount_in_ref,
                         donor_id: self.organization_id,
                     })
-                    posting_items << credit
-              
+                    posting_items << credit              
 
 
 
                 post( Posting.document_types[:project], self.id, Posting.posting_types[:normal], posting_items)
             
             
-            end
     
+    
+    
+        elsif(self.is_a?(StockTake))
+             statistics_account = Account.find_by({'code': :statistics})
+             stock_account = Account.find_by({'code': :stock})
+             lost_account = Account.find_by({'code': :lost})
+
+             stock_take_journal =  Journal.find_by({'code': :annual_inventory})  
+             posting_items = []     
+             self.adjustments.each do |adjustment|
+               puts "posting -----#{adjustment.adjustment_type}   #{Adjustment.adjustment_types[:loss]}"
+                if adjustment.loss? 
+                    
+                    debit = PostingItem.new({
+                            account_id: stock_account.id,
+                            journal_id: stock_take_journal.id,
+                            commodity_id: adjustment.commodity_id,
+                            commodity_category_id: adjustment.commodity_category_id,
+                            quantity: adjustment.amount,
+                            hub_id: adjustment.stock_take.hub_id,
+                            warehouse_id: adjustment.stock_take.warehouse_id,
+                            store_id: adjustment.stock_take.store_no,
+                            donor_id: adjustment.stock_take.donor_id                                                
+
+                        })
+
+                        posting_items << debit
+                        credit = PostingItem.new({
+                            account_id: stock_account.id,
+                            journal_id: stock_take_journal.id,
+                            commodity_id: adjustment.commodity_id,
+                            commodity_category_id: adjustment.commodity_category_id,
+                            quantity: -adjustment.amount,
```

Comment:
This is due to a previous approach where loss and gain were interpreted according to the amount sign in the adjustment record -ve for loss and +ve for gain so that on the postings -amount will be positive and +amount will be positive. But this is confusing so i have added adjustment_type field on adjustment model which tells us if it loss/gain while amount always stays positive.  
The postings will be updated accordingly.